src/tr/tr.rs

```diff
@@ -73,21 +80,20 @@ fn tr<'a>(set1: ExpandSet<'a>, mut set2: ExpandSet<'a>) {
         map.insert(i as usize, s2_prev);
     }
 
-    let mut reader = BufReader::new(stdin());
-    let mut writer = BufWriter::new(stdout);
+    stdin.lock();
 
-    while let Ok(length) = reader.read_to_string(&mut buf) {
+    while let Ok(length) = stdin.read_line(&mut buf) {
         if length == 0 { break }
 
-        {
-            let mut chars = buf.chars();
-
-            while let Some(ch) = chars.next() {
-                let trc = match map.get(&(ch as usize)) {
+        { // isolation to make borrow checker happy
+            let output_stream = buf.chars()
+               .map(|c| match map.get(&(c as usize)) {
                     Some(t) => *t,
-                    None => ch,
-                };
-                safe_unwrap!(writer.write_all(format!("{}", trc).as_ref()));
+                    None => c,
+                });
```

Comment:
Could be rewritten as `buf.chars().map(|c| *map.get(&(c as usize)).unwrap_or(&c))`